keg_storage/__init__.py

```diff
@@ -0,0 +1,48 @@
+import collections
+
+
+class Storage:
+    """A proxy and management object for storage backends."""
+
+    _interaces = None
+
+    def __init__(self, app=None):
+
+        if app:
+            self.init_app(app)
+
+    def init_app(self, app):
+
+        self.app = app
+        if self.app is None:
+            raise AttributeError('app can not be none')
+
+        self._interfaces = collections.OrderedDict(
+            (params['name'], interface(**params))
+            for interface, params in app.config['STORAGE_PROFILES']
+        )
+
+        self.interface = next(iter(self._interfaces)) if self._interfaces else None
+
+    def get_interface(self, interface=None):
+        return self._interfaces[interface or self.interface]
+
+
+class StorageBackend:
```

Comment:
@nZac I wanted to do that, but since you didn't do it I figured I'd not rock the boat. I have a tendency to rock the boat too much. ;)